### ISSUE TITLE
Refactoring commands are not (always) working correctly

### ISSUE THREAD

**Issue by andrzejkupczyk** [vimeo/psalm#4533]
Psalm refactoring tool breaks the code. This happens when you try to rename a class which already extends another class with the same name as the new one. For example, running `vendor/bin/psalm-refactor --rename "Ns1\Foo" --to "Ns2\Bar\Baz"` changes original content of a file:

```php
<?php

namespace Ns1;

class Foo extends Baz {
    public function qux(): void {}
}
```

into something like this:
```php
Baz?php

namespace Ns2\Bar;

class Baz extends Baz
{
    public function qux(): void {}
}
void
```

**Comment by andrzejkupczyk** [comment]
Well, no :) This issue regards the refactoring tool.

**Comment by muglug** [comment]
Can you reproduce in a separate repo? Or even better, [in a test](https://github.com/vimeo/psalm/blob/master/tests/FileManipulation/ClassMoveTest.php)?

**Comment by andrzejkupczyk** [comment]
@muglug , I really tried, but was not able reproduce this behaviour in a test :thinking:   
This is so weird… I was quite sure that my test case will fail (https://github.com/andrzejkupczyk/psalm/commit/d93b85ee2bd735273a09a3237d58c0707f50569c) since it does not work with my code :exploding_head:

**Comment by muglug** [comment]
> The only thing that came to my mind is that maybe the test itself has some flaw?

I don't believe that's the issue. To confirm you could create a separate repository and confirm it doesn't work.

You could start by reducing the file you're trying to modify down to its most basic components to see if you still get the bug.

**Comment by andrzejkupczyk** [comment]
Okay, I manage to reproduce this in [a separate repository](https://github.com/andrzejkupczyk/psalm-issue). This is very simple code.  
Please take a look and tell me that this works for you as expected :slightly_smiling_face: 

## Steps 

```bash
$ git clone https://github.com/andrzejkupczyk/psalm-issue
$ cd psalm-issue
$ composer install
$ vendor/bin/psalm-refactor --move "Ns1\Ns2\LaravelServiceProvider" --into "Ns1\Ns2\Laravel\ServiceProvider" 
```

## Result

https://raw.githubusercontent.com/andrzejkupczyk/psalm-issue/result/src/Laravel/ServiceProvider/LaravelServiceProvider.php

**Comment by andrzejkupczyk** [comment]
I've just discovered that the refactoring tool works correctly if you remove the `illuminate/support` dependency :see_no_evil:

**Comment by andrzejkupczyk** [comment]
@muglug , please let me know if there is more details that I could provide. I'm always happy to help.
